astropy/units/core.py

```diff
@@ -2116,10 +2103,7 @@ def __call__(
                 if parse_strict != "silent":
                     # Deliberately not issubclass here. Subclasses
                     # should use their name.
-                    if f is not unit_format.Generic:
-                        format_clause = f.name + " "
-                    else:
-                        format_clause = ""
+                    format_clause = "" if f is unit_format.Generic else f.name + " "
```

Comment:
Could even do this inside `msg`, but an improvement as is.

Reply:
The main reason for keeping this separate was to let it retain its comment.